src/ScriptEditor/ui/Tabs.tsx

```diff
@@ -1,20 +1,24 @@
 import React, { useState } from "react";
 import { DragDropContext, Droppable, Draggable } from "react-beautiful-dnd";
```

Comment:
```suggestion
import { DragDropContext, Droppable, Draggable, DropResult } from "react-beautiful-dnd";
```
Import needed for suggested onDragEnd type fix

Reply:
@Snarling I think it might look better this way:
```tsx
  const filteredScripts = Object.values(scripts)
    .map((script, originalIndex) => ({ script, originalIndex }))
    .filter(({ script }) => script.hostname.includes(filter) || script.path.includes(filter));

  function onDragEnd(result: DropResult): void {
    // Dropped outside of the list
    if (!result.destination) return;
    reorder(
      scripts,
      filteredScripts[result.source.index].originalIndex,
      filteredScripts[result.destination.index].originalIndex,
    );
    rerender();
  }
  ...
             {filteredScripts.map(({ script, originalIndex }, index) => {
                const { path: fileName, hostname } = script;
                const isActive = currentScript?.path === script.path && currentScript.hostname === script.hostname;

                const title = `${hostname}:~${fileName.startsWith("/") ? "" : "/"}${fileName} ${dirty(scripts, index)}`;

                return (
                  <Draggable
                    ...
                    index={index}
                  >
                    {(provided) => (
                      <Tab
                       ...
                        onClick={() => onTabClick(originalIndex)}
                        onClose={() => onTabClose(originalIndex)}
                        onUpdate={() => onTabUpdate(originalIndex)}
                      />
                    )}
                  </Draggable>
                );
              })}
```
wdyt?

Reply:
As for the tooltip issue - I'll take a look into it. Most probably it's not specific to script tabs and should be reproducible in other places with similar component behavior.

Reply:
https://github.com/mui/material-ui/issues/12299

Reply:
@Snarling should be fixed now, could you please check? :)

Reply:
Yeah your rewrite works a lot better. Tooltip workaround also seems to work, I couldn't trigger any unexpected behavior. Good stuff!